StyleCop.Analyzers/StyleCop.Analyzers.Test/LayoutRules/SA1518UnitTests.cs

```diff
@@ -0,0 +1,152 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using StyleCop.Analyzers.LayoutRules;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using TestHelper;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.CodeFixes;
+
+namespace StyleCop.Analyzers.Test.LayoutRules
+{
+    /// <summary>
+    /// Unit tests for the SA1518: CodeMustNotContainBlankLinesAtEndOfFile
+    /// </summary>
+    [TestClass]
+    public class SA1518UnitTests : CodeFixVerifier
+    {
+        private const string DiagnosticId = SA1518CodeMustNotContainBlankLinesAtEndOfFile.DiagnosticId;
+
+        private const string BaseCode = @"using System.Diagnostics;
+public class Foo
+{
+    public void Bar(int i)
+    {
+        Debug.Assert(true);
+    }
+}";
+        
+        /// <summary>
+        /// Verifies that the analyzer will properly handle an empty source.
+        /// </summary>
+        /// <returns></returns>
+        [TestMethod, TestCategory("MaintainabilityRules/SA1518")]
```

Comment:
No problem!
I think adding these to all other tests would make testing easier. You can group by class in the test explorer which would group the same tests together as well, but you don't see the category there so it is not as clear. I whish you could apply the TestCategory attribute to classes but unfortunatly thats not possible.